src/plot_api/plot_api.js

```diff
@@ -956,6 +956,12 @@ Plotly.addTraces = function addTraces(gd, traces, newIndices) {
     if(!Array.isArray(traces)) {
         traces = [traces];
     }
+
+    // make sure traces do not repeat existing ones
+    traces = traces.map(function(trace) {
+        return Lib.extendFlat({}, trace);
```

Comment:
Desired behavior confirmed in https://github.com/plotly/plotly.js/issues/1083#issuecomment-260396248